lib/api/units/validateHeaders.js

```diff
@@ -0,0 +1,68 @@
+const { HeadersJsonExample } = require('../../validators/headers-json-example');
+
+const APIARY_JSON_HEADER_TYPE = 'application/vnd.apiary.http-headers+json';
+
+function normalizeHeaders(headers) {
+  return (
+    headers instanceof Object &&
```

Comment:
what happens if this is `false`? the function returns `false`?

Reply:
Yes. I guess it would be more obvious to return explicit `null` in that case. What do you think?

Reply:
Returning `null` in e058feb4869ec53257a6f372e9e71b409fcc3985

Reply:
I think the headers should come in as an object. If they're not, we should:

- coerce it to `{}` if we think it is a valid situation
- throw a runtime error if we think it is an invalid situation

I don't see a good reason why the rest of Gavel should call this function with anything else than an object, so I'm inclined to the second option. We should make sure there are clear interfaces with expectable types. If the surrounding code doesn't play along, let's change the surrounding code so it does. E.g. if Gavel takes whatever user passes into it and gives it to `validateHeaders()` as it is, perhaps it should rather make sure headers are either full of headers or default to an empty object. Or, if user passes rubbish, then Gavel should halt and crash with an error.